### ISSUE TITLE
Selector waits eternally for elements to appear when the date is mocked by using client scripts.

### ISSUE THREAD

**Issue by Thormod** [DevExpress/testcafe#5447]
### What is your Test Scenario?

For my Integration tests, I need to mock the browser date and to accomplish that I'm using `clientScripts` injecting the `mockdate` module:

```javascript
const clientScriptsObject = [
  { module: 'mockdate' },
  { content: "MockDate.set('2020-04-19')" },
];
```
For this issue, I created a small project that goes to Wikipedia and tries to search for something but using an incorrect selector:
```javascript
const mockDate = `
    Date.now = function () {
        return 1566408247;
    };
`;

fixture('My first fixture').page('https://www.wikipedia.org/');

test('users can search', async (t) => {
  await t
    .click(Selector('#searchInput'))
    .click(Selector('#this-selector-does-not-exist'))
    .typeText(Selector('#searchInput'), 'vikings')
    .click(Selector('button[type=submit]'));
}).clientScripts({ content: mockDate });
 ```
### What is the Current behavior?

 When the tests execute ` .click(Selector('#this-selector-does-not-exist'))` the `Selector` get stuck waiting for the element to appear, ignoring the default (and setted) selector timeouts:

![image](https://user-images.githubusercontent.com/7811197/90932366-1e51e480-e3c4-11ea-82e0-bf891797d13f.png)

### What is the Expected behavior?

To raise an error specifying that the selector does not exist.

![image](https://user-images.githubusercontent.com/7811197/90932418-3a558600-e3c4-11ea-8fbe-bd3932b5f447.png)
 
 
### What is your web application and your TestCafe test code?

You can access the code here:
https://github.com/Thormod/testcafe-issue
 
<details>
<summary>Your website URL (or attach your complete example):</summary>

</details>

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
 
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>

```

```
</details>

<details>
<summary>Your complete test report:</summary>

```

```
</details>

<details>
<summary>Screenshots:</summary>

```

```
</details>

### Steps to Reproduce:

Just run `npm run test` on the example project, or mock the `Date.now()` function by using `clientScript`
 
### Your Environment details:
 
* testcafe version:                   ^1.9.1
* node.js version:                    v10.19.0
* command-line arguments:  `testcafe chrome`
* browser name and version: Chrome latest
* platform and version:           MacOs Catalina 10.15.6

**Comment by miherlosev** [comment]
Hi @Thormod

Thank you for the detailed information. I've reproduced the problem.

**Comment by Thormod** [comment]
@miherlosev Do you know when the new version will be released? PD: Thank you for solving this bug!

**Comment by AndreyBelym** [comment]
A release candidate version `testcafe@1.9.2-rc.1` is already published to NPM. You can test it in your environment if you do not mind installing preliminary versions. Once we test it and create a changelog, we will publish a stable release.

**Comment by Thormod** [comment]
@AndreyBelym @miherlosev There are some TypeErrors in the new version:
![image](https://user-images.githubusercontent.com/7811197/91877957-4bb05500-ec44-11ea-945a-d65d24a1e332.png)

```javascript
Uncaught TypeError: Cannot read property 'destUrl' of null
    at Function.t._sameOriginCheck (worker-hammerhead.js:1)
    at r.fetch (worker-hammerhead.js:1)
    at u (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2)
    at b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2
    at xt (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2)
    at Object.bt [as getArrayBuffer] (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2)
    at l.s [as loadVectorData] (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2)
    at l.loadTile (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2)
    at ke.loadTile (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:2)
    at __call$ (b54a5bd5-05e5-4a32-97e0-f1d459ace1fb:1)
```

It seems to be related with the fetch requests:
![image](https://user-images.githubusercontent.com/7811197/91878147-7ef2e400-ec44-11ea-8e86-4481e5c2ee33.png)

**Comment by AndreyBelym** [comment]
I cannot reproduce the issue with Wikipedia.org. But I created my own example, which demonstrates a very similar issue: https://github.com/DevExpress/testcafe-hammerhead/issues/2428. The version with the fix is already published to NPM as `testcafe@1.9.2-rc.2`. As my example and the fix is based on assumptions only, there is a risk that it doesn't fix your specific issue. Could you please try to install `testcafe@1.9.2-rc.2` and provide a sample test page if you still can reproduce the issue in your environment?

**Comment by Thormod** [comment]
@AndreyBelym It worked! Thank u for solving this so fast!
